src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,14 @@
+import Popup from './Popup.js';
+
+export default class PopupWithConfirm extends Popup {
+    constructor(popupSelector, handleDeleteCard) {
+        super({ popupSelector });
+        this._confirmButton = document.querySelector('#delete-card');
+        this._handleDeleteCard = handleDeleteCard;
```

Comment:
`PopupWithConfirmation` is a general class. It can be used for any confirmation, so there should not be mentions of `card`. Also, there should be only one instance of this class, so you should add a setter for the submit handler.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._confirmButton = ..
}

setAction(action) { 
  this._handleConfirm = action; 
} 

..
this._confirmButton.addEventListener('click', this._handleConfirm); 

```

Reply:
I've managed to reuse the Popup class for the confirmation functionality... and it works! But, is that good practice?

Reply:
There should be `PopupWithConfirmation` class, but it should be general (without mentions of the cards). It should extend the popup class, have a method for setting the submit action and handle the submission

```js
class PopupWithConfirmation extends Popup {
  ..
  setAction(action) {
    this._handleConfirm = action; 
  }
}
```